src/Bicep.LangServer.IntegrationTests/CodeActionTestBase.cs

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading;
+using Bicep.Core.CodeAction;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Extensions;
+using Bicep.Core.Parsing;
+using Bicep.Core.Samples;
+using Bicep.Core.Syntax;
+using Bicep.Core.Text;
+using Bicep.Core.UnitTests;
+using Bicep.Core.UnitTests.Assertions;
+using Bicep.Core.UnitTests.PrettyPrintV2;
+using Bicep.Core.UnitTests.Serialization;
+using Bicep.Core.UnitTests.Utils;
+using Bicep.Core.Workspaces;
+using Bicep.LangServer.IntegrationTests.Helpers;
+using Bicep.LanguageServer.Extensions;
+using Bicep.LanguageServer.Utils;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Newtonsoft.Json.Linq;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Client;
+using OmniSharp.Extensions.LanguageServer.Protocol.Document;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+
+namespace Bicep.LangServer.IntegrationTests
+{
+    [TestClass]
+    public class CodeActionTestBase
+    {
+        private static SemaphoreSlim initialize = new(1);
+        private static bool isInitialized = false;
+
+        protected static ServiceBuilder Services => new();
+
+        protected static readonly SharedLanguageHelperManager DefaultServer = new();
+
+        protected static readonly SharedLanguageHelperManager ServerWithFileResolver = new();
+
+        protected static readonly SharedLanguageHelperManager ServerWithBuiltInTypes = new();
+
+        protected static readonly SharedLanguageHelperManager ServerWithNamespaceProvider = new();
+
+        [NotNull]
+        public TestContext? TestContext { get; set; }
+
+        [ClassInitialize(InheritanceBehavior.BeforeEachDerivedClass)]
+        public static void ClassInitialize(TestContext testContext)
+        {
+            initialize.Wait();
+            try
+            {
+                if (!isInitialized)
+                {
+                    isInitialized = true;
+                    DefaultServer.Initialize(async () => await MultiFileLanguageServerHelper.StartLanguageServer(testContext));
+                    ServerWithFileResolver.Initialize(async () => await MultiFileLanguageServerHelper.StartLanguageServer(testContext));
+                    ServerWithBuiltInTypes.Initialize(async () => await MultiFileLanguageServerHelper.StartLanguageServer(testContext, services => services.WithNamespaceProvider(BuiltInTestTypes.Create())));
+                    ServerWithNamespaceProvider.Initialize(async () => await MultiFileLanguageServerHelper.StartLanguageServer(testContext, services => services.WithNamespaceProvider(BicepTestConstants.NamespaceProvider)));
```

Comment:
nit: can elide `async` and `await` since it's just a passthrough.

Reply:
Test code, I just moved it, not worth the effort to change.